src/lib.rs

```diff
@@ -89,6 +89,59 @@ impl<K: std::hash::Hash + PartialEq, V> HashTable<K, V> {
     }
 }
 
+pub struct HashTableIterator<'a, K, V> {
+    current_bucket: Option<&'a Vec<(K, V)>>,
+    elements_iterator: Box<dyn Iterator<Item = &'a (K, V)> + 'a>,
```

Comment:
I am not confident about using the lifetime on the `Item` type. I did it as a result of trial and error, but I don't really understand it.

Reply:
The lifetime constraint is on the Iterator which makes more sense since the compiler complains that inferring a lifetime is not possible with respect to calls to `iter` on the `Iterator`s. I still don't get what it means though. 
Something related: https://users.rust-lang.org/t/lifetime-errors/4130